core/src/main/java/net/sourceforge/jnlp/tools/JarCertVerifier.java

```diff
@@ -351,22 +351,22 @@ VerifyResult verifyJarEntryCerts(final String jarPath, final boolean jarHasManif
                     final ZonedDateTime notBefore = zonedDateTime(((X509Certificate) cert).getNotBefore());
                     final ZonedDateTime notAfter = zonedDateTime(((X509Certificate) cert).getNotAfter());
 
-                    final Optional<Timestamp> optionalSignatureTimestamp = Optional.ofNullable(codeSigners.get(certPath))
+                    final Optional<Timestamp> optionalTsa = Optional.ofNullable(codeSigners.get(certPath))
                             .map(CodeSigner::getTimestamp);
-                    final X509Certificate tsaCertificate = (X509Certificate) optionalSignatureTimestamp
+                    final X509Certificate tsaCertificate = (X509Certificate) optionalTsa
                             .map(Timestamp::getSignerCertPath)
                             .map(CertPath::getCertificates)
                             .filter(certs -> !certs.isEmpty())
                             .map(certs -> certs.get(0))
                             .filter(c -> c instanceof X509Certificate)
                             .orElse(null);
 
-                    if (tsaCertificate != null) {
+                    if (tsaCertificate != null && isTrustedTsa(optionalTsa.map(Timestamp::getSignerCertPath).orElseThrow(() -> new RuntimeException("cert path is null even tough we have a tsa certificate")))) {
```

Comment:
Maybe it makes sense to extract the optionalTsa before this if since we use it 2 times:

```
final Timestamp tsa = optionalTsa.orElseThrow(() -> new RuntimeException("...))
if(tsaCertificate != null && isTrustedTsa(tsa)) {
    // ...
    final ZonedDateTime signedAt = zonedDateTime(tsa.getTimestamp());
    //...
}
```

Reply:
I could not do the `orElseThrow` since it is absolutely OK for the optional to be empty in case that `tsaCertificate` is null

Reply:
maybe `.orElse(null)` ?

Reply:
then I think it would almost be simpler to not use the optional...